eth/query/query_test.go

```diff
@@ -233,3 +231,147 @@ func testGetLogs(t *testing.T, v handler.ReceiptHandlerVersion) {
 
 	require.NoError(t, receiptHandler.Close())
 }
+
+func TestDupEvmTxHash(t *testing.T) {
+	blockTxHash := getRandomTxHash()
+	txHash1 := getRandomTxHash() // DeployEVMTx that has dup EVM Tx Hash
+	txHash2 := getRandomTxHash() // CallEVMTx that has dup EVM Tx Hash
+	txHash3 := getRandomTxHash() // DeploEVMTx that has unique EVM Tx Hash
+	txHash4 := getRandomTxHash() // CallEVMTx that has unique EVM Tx Hash
+	from := loom.MustParseAddress("default:0x7262d4c97c7B93937E4810D289b7320e9dA82857")
+	to := loom.MustParseAddress("default:0x7262d4c97c7B93937E4810D289b7320e9dA82857")
+
+	deployTx, err := proto.Marshal(&vm.DeployTx{
+		VmType: vm.VMType_EVM,
+	})
+	callTx, err := proto.Marshal(&vm.DeployTx{
+		VmType: vm.VMType_EVM,
+	})
+
+	signedDeployTxBytes := mockSignedTx(t, deployId, to, from, deployTx)
+	signedCallTxBytes := mockSignedTx(t, callId, to, from, callTx)
+
+	blockResultDeployTx := &ctypes.ResultBlock{
+		BlockMeta: &ttypes.BlockMeta{
+			BlockID: ttypes.BlockID{
+				Hash: blockTxHash,
+			},
+		},
+		Block: &ttypes.Block{
+			Data: ttypes.Data{
+				Txs: ttypes.Txs{signedDeployTxBytes},
+			},
+		},
+	}
+
+	blockResultCallTx := &ctypes.ResultBlock{
+		BlockMeta: &ttypes.BlockMeta{
+			BlockID: ttypes.BlockID{
+				Hash: blockTxHash,
+			},
+		},
+		Block: &ttypes.Block{
+			Data: ttypes.Data{
+				Txs: ttypes.Txs{signedCallTxBytes},
+			},
+		},
+	}
+
+	evmAuxStore, err := common.NewMockEvmAuxStore()
+	require.NoError(t, err)
+
+	dupEVMTxHashes := make(map[string]bool)
+	dupEVMTxHashes[string(txHash1)] = true
+	dupEVMTxHashes[string(txHash2)] = true
+	evmAuxStore.SetDupEVMTxHashes(dupEVMTxHashes)
+
+	txResultData1 := mockDeployResponse(txHash1)
+	txResultData2 := txHash2
+	txResultData3 := mockDeployResponse(txHash3)
+	txResultData4 := txHash4
+
+	// txhash1 is dup, so the returned hash must not be equal
+	txObj, _, err := GetTxObjectFromBlockResult(blockResultDeployTx, txResultData1, int64(0), evmAuxStore)
+	require.NoError(t, err)
+	require.NotEqual(t, string(txObj.Hash), string(eth.EncBytes(txHash1)))
```

Comment:
Should also check that `txObject.Hash` is the same as `types.Tx(signedDeployTxBytes).Hash()`

Reply:
Added